common/tmx_level.go

```diff
@@ -87,17 +84,13 @@ func (t ByFirstgid) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 func (t ByFirstgid) Less(i, j int) bool { return t[i].Firstgid < t[j].Firstgid }
 
 // MUST BE base64 ENCODED and COMPRESSED WITH zlib!
-func createLevelFromTmx(r Resource) (*Level, error) {
+func createLevelFromTmx(tmx, tmxUrl string) (*Level, error) {
 	tlvl := &TMXLevel{}
 	lvl := &Level{}
 
-	tmx, err := readTmx(r.url)
-	if err != nil {
-		return lvl, err
-	}
-
 	if err := xml.Unmarshal([]byte(tmx), &tlvl); err != nil {
 		fmt.Printf("Error unmarshalling XML: %v", err)
+		return lvl, err
```

Comment:
Idiomatic would be to `return nil, err` - consumes less resources and indicates that whatever `*Level` is being returned on failure, is useless.